services/minespace-web/src/components/pages/Project/MajorMineApplicationPage.js

```diff
@@ -0,0 +1,146 @@
+import React, { Component } from "react";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { Link, withRouter } from "react-router-dom";
+import { change, submit, getFormSyncErrors, reset, touch } from "redux-form";
+import { Row, Col, Typography } from "antd";
+import { ArrowLeftOutlined } from "@ant-design/icons";
+import PropTypes from "prop-types";
+import customPropTypes from "@/customPropTypes";
+import { flattenObject } from "@common/utils/helpers";
+import * as FORM from "@/constants/forms";
+import { EDIT_PROJECT } from "@/constants/routes";
+import { getProject } from "@common/reducers/projectReducer";
+import { getMines } from "@common/selectors/mineSelectors";
+import {
+  fetchProjectById,
+  createMajorMineApplication,
+} from "@common/actionCreators/projectActionCreator";
+import { clearMajorMinesApplication } from "@common/actions/projectActions";
+import MajorMineApplicationForm from "@/components/Forms/projects/majorMineApplication/MajorMineApplicationForm";
+import { getMajorMinesApplicationDocumentTypesHash } from "@common/selectors/staticContentSelectors";
+
+const propTypes = {
+  mines: PropTypes.arrayOf(customPropTypes.mine).isRequired,
+  project: customPropTypes.project.isRequired,
+  clearMajorMinesApplication: PropTypes.func.isRequired,
+  createMajorMineApplication: PropTypes.func.isRequired,
+  fetchProjectById: PropTypes.func.isRequired,
+  majorMinesApplicationDocumentTypesHash: PropTypes.objectOf(PropTypes.string).isRequired,
+  match: PropTypes.shape({
+    params: {
+      projectGuid: PropTypes.string,
+    },
+  }).isRequired,
+  touch: PropTypes.func.isRequired,
+  submit: PropTypes.func.isRequired,
+  formErrors: PropTypes.objectOf(PropTypes.string),
+};
+
+const defaultProps = {
+  formErrors: {},
+};
+
+export class MajorMineApplicationPage extends Component {
+  componentDidMount() {
+    this.handleFetchData();
+  }
+
+  componentWillUnmount() {
+    this.props.clearMajorMinesApplication();
+  }
+
+  handleFetchData = () => {
+    const { projectGuid } = this.props.match?.params;
+    return this.props.fetchProjectById(projectGuid);
+  };
+
+  handleCreateMajorMineApplication = (values, message) => {
+    const fileUploads = document.getElementsByClassName("filepond--action-revert-item-processing");
```

Comment:
Could we add a comment above this variable describing which part of the Filepond component we are clicking and why? Just thinking about future devs coming across this function and perhaps being a bit confused.

Reply:
Thank you. Going to reuse the common function from minespace-web/common/utils/helpers.js:
export const cleanFilePondFile = () => {
  const fileUploaded = document.getElementsByClassName("filepond--action-revert-item-processing");
  if (fileUploaded.length > 0) {
    fileUploaded.forEach((file) => file.click());
  }
};